cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -73,6 +76,15 @@
   private static final String TYPE_NAMESPACE = "namespace";
   private static final String TYPE_ADAPTER = "adapter";
   private static final String WORKFLOW_TOKEN_PROPERTY_KEY = "workflowToken";
+  private static final int MDS_SCAN_LIMIT = 5000;
```

Comment:
Good point! Made the scan limit half of transaction timeout instead of number of elements.